executor/set_test.go

```diff
@@ -1568,3 +1569,20 @@ func TestSetTopSQLVariables(t *testing.T) {
 	tk.MustQuery("show variables like '%top_sql%'").Check(testkit.Rows("tidb_enable_top_sql OFF", "tidb_top_sql_max_meta_count 5000", "tidb_top_sql_max_time_series_count 20"))
 	tk.MustQuery("show global variables like '%top_sql%'").Check(testkit.Rows("tidb_enable_top_sql OFF", "tidb_top_sql_max_meta_count 5000", "tidb_top_sql_max_time_series_count 20"))
 }
+
+func TestInstanceScopeSwitching(t *testing.T) {
+
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	// enable 'switching' to SESSION variables
+	tk.MustExec("set tidb_enable_legacy_instance_scope = 1")
+	tk.MustExec("set tidb_general_log = 1")
+	tk.MustQuery(`show warnings`).Check(testkit.Rows(fmt.Sprintf("Warning %d allowing variable tidb_general_log with INSTANCE scope change to SESSION scope while setting for now, but its deprecated", errno.ErrInstanceScope)))
```

Comment:
This line will require changing for the updated error message

Reply:
Thanks!

It seems that `check_dev` fails due to the introduction of the new error code:

```text
[2022-03-10T06:10:56.188Z] Files errors.toml and /tmp/errors.toml.before differ

[2022-03-10T06:10:56.188Z] make: *** [errdoc] Error 1
```